libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -830,6 +844,9 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 							errinfo_comment("Stack too deep, try removing local variables.")
 						);
 					_assembly.appendInstruction(dupInstruction(stackDiff));
+					if (leftShiftSelector)
```

Comment:
Ok, so it is right-aligned on the stack? Then we should maybe also just return it like that.